### DISCUSSION TITLE
Is TypeScript support in Bazel for Windows, or not?

### DISCUSSION THREAD

**Discussion by hagbard** [bazelbuild/rules_nodejs#3671] [Q&A]
I've spent most of a day trying to untangle information about how to get a basic "Hello World" TypeScript build working with a recent release of these rules, but have found myself going round in circles.

Eventually I stumbled on:
https://github.com/bazelbuild/rules_nodejs/wiki/Migrating-to-5.0#ts_library
and:
https://github.com/bazelbuild/rules_nodejs/wiki/Migrating-to-5.0#reduced-maintenance-scope

which says:
1. The `ts_library` rule is now in the `@bazel/concatjs` package.
2. The concatjs package is now in "maintenance mode". Since no one from Google participates in rules_nodejs, it has been impossible for us to keep this code up-to-date with what Google develops internally, and it is a complex package that OSS maintainers don't have time for. In the future we will probably deprecate this package unless new maintainers want to take it over.

which makes me worried that TypeScript (even it is is supported now) won't be supported in future.

So my question is:
1. Is TypeScript support something I can expect to be able to rely on in Bazel in Windows?
2. If TypeScript is supported, what are the instructions for getting it working?

**Comment by alexeagle** [comment]
I'm not sure how you got stuck in that loop, but I'd suggest https://github.com/aspect-build/rules_ts for Bazel/Typescript.
It's mentioned in the deprecation message on https://www.npmjs.com/package/@bazel/typescript
It should work fine on Windows, we have CI testing for that.
